webpack/assets/javascripts/react_app/redux/API/APIMiddleware.js

```diff
@@ -1,10 +1,23 @@
 import { API_OPERATIONS, actionTypeGenerator } from './';
 import { get } from './APIRequest';
+import { startPolling } from './APIActions';
+import { selectPollingID } from './APISelectors';
+import { registeredPollingException } from './APIHelpers';
 
 export const APIMiddleware = store => next => action => {
-  const { type, key, payload = {}, url, actionTypes = {} } = action;
+  const { type, key, payload = {}, url, actionTypes = {}, polling } = action;
   if (type === API_OPERATIONS.GET) {
-    get(payload, url, store, actionTypeGenerator(key, actionTypes));
+    const APIRequest = () =>
+      get(payload, url, store, actionTypeGenerator(key, actionTypes));
+    if (polling) {
+      const isAlreadyPolling = selectPollingID(store.getState(), key);
+      if (isAlreadyPolling) {
+        throw registeredPollingException(key);
+      }
+      next(startPolling(key, APIRequest, polling));
+    } else {
+      APIRequest();
```

Comment:
don't we need to use `next` here?
AFAIK if we will have another  middleware it won't be chained and it will be lost